File: src/ProjectR.Scripting/content/scripts/spells/MagicDampeningZone.lua
function GetTargetType()
    return Enemies
end

function GetName()
return "Magic-Dampening Zone"
end

function GetDescription()
return "Summon a magic-dampening zone.\nReduces enemies MD."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .6
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { ARC }
end

function SpellEffect()
  local damage = (3.0625 * aMD * (aARC/100) - .875 * dMR) * (100/dARC)  
  d:BuffStat(MD, -.18)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/MagicDampeningZone.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class MagicDampeningZone : SpellScriptBase
    {
        public override string Name { get { return "Magic Dampening Zone"; } }
        public override string Description { get { return "Summon a magic dampening zone.\nReduces enemies MD."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ARC, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 60; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (3.0625 * cMD * (cARC / 100) - .875 * tMR) * (100 / tARC);
            BuffStat(Stat.MD, -.38);
            DealDamage(damage);
        }
    }
}